src/StreamJsonRpc/NerdbankMessagePackFormatter.cs

```diff
@@ -0,0 +1,2301 @@
+
+    private readonly PipeConverterResolver pipeConverterResolver;
+
+    private readonly MessagePackExceptionResolver exceptionResolver;
+
+    private readonly ToStringHelper serializationToStringHelper = new();
+
+    private readonly ToStringHelper deserializationToStringHelper = new();
+
+    /// <summary>
+    /// The serializer to use for user data (e.g. arguments, return values and errors).
+    /// </summary>
+    private FormatterContext userDataContext;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NerdbankMessagePackFormatter"/> class.
+    /// </summary>
+    public NerdbankMessagePackFormatter()
+    {
+        // Set up initial options for our own message types.
+        NBMP::MessagePackSerializer serializer = new()
+        {
+            InternStrings = true,
+            SerializeDefaultValues = false,
+        };
+
+        serializer.RegisterConverter(new RequestIdConverter());
+        serializer.RegisterConverter(new JsonRpcMessageConverter(this));
+        serializer.RegisterConverter(new JsonRpcRequestConverter(this));
+        serializer.RegisterConverter(new JsonRpcResultConverter(this));
+        serializer.RegisterConverter(new JsonRpcErrorConverter(this));
+        serializer.RegisterConverter(new JsonRpcErrorDetailConverter(this));
+        serializer.RegisterConverter(new TraceParentConverter());
+
+        this.rpcContext = new FormatterContext(serializer, ShapeProvider_StreamJsonRpc.Default);
+
+        // Create the specialized formatters/resolvers that we will inject into the chain for user data.
+        this.progressConverterResolver = new ProgressConverterResolver(this);
+        this.asyncEnumerableConverterResolver = new AsyncEnumerableConverterResolver(this);
+        this.pipeConverterResolver = new PipeConverterResolver(this);
+
+        // TODO: Convert these to converter resolvers?
+        this.exceptionResolver = new MessagePackExceptionResolver(this);
+
+        FormatterContext userDataContext = new(
+            new()
+            {
+                InternStrings = true,
+                SerializeDefaultValues = false,
+            },
+            ReflectionTypeShapeProvider.Default);
+
+        this.MassageUserDataContext(userDataContext);
+        this.userDataContext = userDataContext;
+    }
+
+    private interface IJsonRpcMessagePackRetention
+    {
+        /// <summary>
+        /// Gets the original msgpack sequence that was deserialized into this message.
+        /// </summary>
+        /// <remarks>
+        /// The buffer is only retained for a short time. If it has already been cleared, the result of this property is an empty sequence.
+        /// </remarks>
+        ReadOnlySequence<byte> OriginalMessagePack { get; }
+    }
+
+    /// <inheritdoc cref="FormatterBase.MultiplexingStream"/>
+    public new MultiplexingStream? MultiplexingStream
+    {
+        get => base.MultiplexingStream;
+        set => base.MultiplexingStream = value;
+    }
```

Comment:
This leaves me wondering why it's here.

Reply:
I really can't see a reason for it. Copied it directly from MessagePackFormatter. Can't seem any harm in removing it.